File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPGliding.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPGliding implements PlayerStateProperty<Boolean> {
    @Override
    public String getConfigName() {
        return "gliding";
    }

    @Override
    public Boolean getDefaultSerializedProperty() {
        return false;
    }

    @Override
    public Boolean getSerializedPropertyFromPlayer(Player player) {
        return player.isGliding();
    }

    @Override
    public void restoreProperty(Player player, Boolean gliding) {
        player.setGliding(gliding);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPGliding.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPGliding extends PlayerStateProperty[Boolean]:
  override def getConfigName: String = "gliding"

  override def getDefaultSerializedProperty: Boolean = false

  override def getSerializedPropertyFromPlayer(player: Player): Boolean =
    player.isGliding

  override def restoreProperty(player: Player, gliding: Boolean): Unit =
    player.setGliding(gliding)